trigger/task_jira.go

```diff
@@ -71,42 +69,63 @@ func getTaskURL(data *jiraTemplateData) (string, error) {
 	return taskLink(data.UIRoot, id, execution), nil
 }
 
-func getTaskLogURL(data *jiraTemplateData) (string, error) {
-	if data.Task == nil {
-		return "", errors.New("task is nil")
-	}
-	id := data.Task.Id
-	execution := data.Task.Execution
-	if len(data.Task.OldTaskId) != 0 {
-		id = data.Task.OldTaskId
-	}
-
-	return taskLogLink(data.UIRoot, id, execution), nil
-}
-
-type executionTaskInfo struct {
+type taskInfo struct {
 	Tests string
 	URL   string
 }
 
-func getExecutionTaskLogURLs(data *jiraTemplateData) []executionTaskInfo {
-	execTaskMap := make(map[string][]jiraTestFailure)
-	for _, test := range data.Tests {
-		execTaskMap[test.TaskID] = append(execTaskMap[test.TaskID], test)
+func getTaskLogURLs(data *jiraTemplateData) ([]taskInfo, error) {
+	if data.Task == nil {
+		return nil, errors.New("task is nil")
 	}
-	result := make([]executionTaskInfo, 0, len(execTaskMap))
-	for taskID, tests := range execTaskMap {
-		info := executionTaskInfo{URL: taskLogLink(data.UIRoot, taskID, tests[0].Execution)}
-		testIDs := make([]string, 0, len(tests))
-		for _, test := range tests {
-			testIDs = append(testIDs, test.Name)
-		}
-		info.Tests = strings.Join(testIDs, " ")
 
-		result = append(result, info)
+	if data.Task.DisplayOnly {
+		// Task is display only with tests
+		if len(data.Tests) > 0 {
+			execTaskMap := make(map[string][]jiraTestFailure)
+			result := make([]taskInfo, 0, len(execTaskMap))
+			for _, test := range data.Tests {
+				execTaskMap[test.TaskID] = append(execTaskMap[test.TaskID], test)
+			}
+			for taskID, tests := range execTaskMap {
+				info := taskInfo{URL: taskLogLink(data.UIRoot, taskID, tests[0].Execution)}
+				testIDs := make([]string, 0, len(tests))
+				for _, test := range tests {
+					testIDs = append(testIDs, test.Name)
+				}
+				info.Tests = strings.Join(testIDs, " ")
+
+				result = append(result, info)
+			}
+			return result, nil
+		} else {
+			// Task is display only without tests
+			result := make([]taskInfo, 0)
+			for _, taskName := range data.Task.ExecutionTasks {
+				execTask, err := task.FindOneId(taskName)
+				if err != nil {
+					return nil, err
+				}
+
+				if execTask.Status == "failed" {
+					id := execTask.Id
+					execution := execTask.Execution
+					displayName := execTask.DisplayName
+					info := taskInfo{Tests: displayName, URL: taskLogLink(data.UIRoot, id, execution)}
+					result = append(result, info)
+				}
+			}
+			return result, nil
+		}
+	} else {
+		// Task is not display only
+		id := data.Task.Id
+		execution := data.Task.Execution
+		if len(data.Task.OldTaskId) != 0 {
+			id = data.Task.OldTaskId
+		}
+		return []taskInfo{{Tests: "Display", URL: taskLogLink(data.UIRoot, id, execution)}}, nil
```

Comment:
Why are we setting Tests equal to "Display"?  I would think this should be the task's display name (although it isn't super clear to me what "Tests" is for)